src/entity_gui.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * This file is part of FreeRCT.
+ * FreeRCT is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * FreeRCT is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with FreeRCT. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file entity_gui.cpp Entity windows. */
+
+#include "entity_gui.h"
+
+/** Widget parts of the #EntityRemoveWindow. */
+static const WidgetPart _entity_remove_widgets[] = {
+	Intermediate(0, 1),
+		Intermediate(1, 0),
+			Widget(WT_TITLEBAR, INVALID_WIDGET_INDEX, COL_RANGE_RED), SetData(GUI_ENTITY_REMOVE_CAPTION, GUI_TITLEBAR_TIP),
+			Widget(WT_CLOSEBOX, INVALID_WIDGET_INDEX, COL_RANGE_RED),
+		EndContainer(),
+		Widget(WT_PANEL, INVALID_WIDGET_INDEX, COL_RANGE_RED),
+			Intermediate(2,0),
+				Widget(WT_CENTERED_TEXT, ERW_MESSAGE, COL_RANGE_RED),
+						SetData(GUI_ENTITY_REMOVE_MESSAGE, STR_NULL), SetPadding(5, 5, 5, 5),
+				EndContainer(),
+			Intermediate(1, 5), SetPadding(0, 0, 3, 0),
+				Widget(WT_EMPTY, INVALID_WIDGET_INDEX, COL_RANGE_INVALID), SetFill(1, 0),
+				Widget(WT_TEXT_PUSHBUTTON, ERW_NO, COL_RANGE_YELLOW), SetData(GUI_ENTITY_REMOVE_NO, STR_NULL),
+				Widget(WT_EMPTY, INVALID_WIDGET_INDEX, COL_RANGE_INVALID), SetFill(1, 0),
+				Widget(WT_TEXT_PUSHBUTTON, ERW_YES, COL_RANGE_YELLOW), SetData(GUI_ENTITY_REMOVE_YES, STR_NULL),
+				Widget(WT_EMPTY, INVALID_WIDGET_INDEX, COL_RANGE_INVALID), SetFill(1, 0),
+			EndContainer(),
+	EndContainer(),
+};
+
+Point32 EntityRemoveWindow::OnInitialPosition()
+{
+	Point32 pt;
+	pt.x = (_video.GetXSize() - this->rect.width ) / 2;
+	pt.y = (_video.GetYSize() - this->rect.height) / 2;
+	return pt;
```

Comment:
FWIW I added operators to Point in Pull Req 136, but that hasn't been merged yet. Since it seems like the project is currently on hold forever, I'm not sure it'll ever be addressed.